pkg/config/validation/validation.go

```diff
@@ -228,6 +230,25 @@ func ValidateHTTPHeaderName(name string) error {
 	if name == "" {
 		return fmt.Errorf("header name cannot be empty")
 	}
+	if !validHeaderRegex.MatchString(name) {
+		return fmt.Errorf("header name %s is not a valid header name", name)
```

Comment:
This is a breaking change?

Reply:
If these are set today they break the traffic at runtime with Envoy anyways, so I think this is probably fine? Usually we are ok with breakage for rare cases that are already broken anyways